platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/IsolatedProjectsToolingApiGradleProjectIntegrationTest.groovy

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configurationcache.isolated
+
+import org.gradle.tooling.model.GradleProject
+
+import static org.gradle.configurationcache.isolated.ToolingModelChecker.checkGradleProject
+
+class IsolatedProjectsToolingApiGradleProjectIntegrationTest extends AbstractIsolatedProjectsToolingApiIntegrationTest {
+
+    def "can fetch GradleProject model for empty projects"() {
+        settingsFile << """
+            rootProject.name = 'root'
+
+            include(":lib1")
+            include(":lib1:lib11")
+        """
+
+        when: "fetching without Isolated Projects"
+        def expectedProjectModel = fetchModel(GradleProject)
+
+        then:
+        fixture.assertNoConfigurationCache()
+
+        with(expectedProjectModel) {
+            it.name == "root"
+            it.tasks.size() > 0
+            it.children.size() == 1
+            it.children[0].children.size() == 1
+        }
+
+        when: "fetching with Isolated Projects"
+        executer.withArguments(ENABLE_CLI)
+        def projectModel = fetchModel(GradleProject)
+
+        then:
+        fixture.assertStateStored {
+            modelsCreated(":", 2) // GradleProject, intermediate IsolatedGradleProjectInternal
```

Comment:
💭  I wonder if we should change the fixture method (in a later PR) so that the models are explicitly listed, instead of using a count and a clarifying comment.

Reply:
I was thinking about the same thing 👍 

https://github.com/gradle/gradle/issues/26998